### PR TITLE
refactor(nexmark): move nexmark generator to a standalone crate

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR moves the nexmark generator from the connector crate to a standalone crate in utils, for better maintainability and reusability.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/connector/src/source/nexmark/mod.rs [hunk 1]
```diff
@@ -12,21 +12,18 @@
...
+use nexmark::config::{NexmarkConfig, RateShape};
 use serde::Deserialize;
 use serde_with::{serde_as, DisplayFromStr};
 pub use split::*;
 
-const NEXMARK_BASE_TIME: usize = 1_436_918_400_000;
-
 pub const NEXMARK_CONNECTOR: &str = "nexmark";
 
 const fn identity_i32<const V: i32>() -> i32 {
```
Comment (BugenZhao) [inline src/connector/src/source/nexmark/mod.rs:29]: Can we directly use `core::convert::identity` for this? 🤣

File: src/utils/nexmark/src/generator.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    /// Return the wall clock base time in ms.
+    pub const fn wall_clock_base_time(&self) -> u64 {
+        self.wall_clock_base_time
+    }
+}
+
+impl Iterator for EventGenerator {
+    type Item = Event;
+
+    fn next(&mut self) -> Option<Event> {
```
Comment (BugenZhao) [inline src/utils/nexmark/src/generator.rs:50]: 🥵

**Review by BugenZhao — APPROVED** [review]
LSTM!!!

File: src/connector/src/source/nexmark/mod.rs (hunk 1 above)
Comment (wangrunji0408) [reply src/connector/src/source/nexmark/mod.rs:29]: Seems not. This utilizes const generic but `identity` doesn't have it.

File: src/utils/nexmark/src/generator.rs (hunk 2 above)
Comment (wangrunji0408) [reply src/utils/nexmark/src/generator.rs:50]: 🥵: Why 🥵?

Comment (BugenZhao) [reply src/utils/nexmark/src/generator.rs:50]: It's cool to have them decoupled.

**Pull request merged**